Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -162,8 +162,8 @@ static bool shouldAllowNonInteractiveCursorForElement(const Element& element)
         return true;
 #endif
 
-    if (is<HTMLTextFormControlElement>(element))
-        return !element.focused();
+    if (auto* textElement = dynamicDowncast<HTMLTextFormControlElement>(element))
```

Comment:
```suggestion
    if (RefPtr textElement = dynamicDowncast<HTMLTextFormControlElement>(element))
```

Reply:
@pxlcoder all other `dynamicDowncast`s in the file use variations of `auto` as type for the return result. Is there any particular reason why mine should be different?

Reply:
We are adopting smart pointers everywhere in WebKit, so we shouldn't introduce new raw pointer usage. The other cases were likely before the policy was announced, or were simply missed during review.

https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines